csharp/src/Microsoft.ML.OnnxRuntime/NativeMethods.cs

```diff
@@ -503,59 +560,200 @@ static NativeMethods()
+        /// <param name="name">model output name (utf-8)</param>
+        /// <param name="ort_value">OrtValue that is used for output (may wrap arbitrary memory). 
+        ///      The param instance is copied internally so this argument may be released.
+        /// </param>
+        public delegate IntPtr /* OrtStatus*/ DOrtBindOutput(IntPtr /*(OrtIoBinding)*/ io_binding, IntPtr /*(const char*) */ name, IntPtr /*const OrtValue**/ ort_value);
+        public static DOrtBindOutput OrtBindOutput;
+
+        /// <summary>
+        /// Bind a device to the model output with the specified name
+        /// This is useful when the OrtValue can not be allocated ahead of time
+        /// due to unknown dimensions.
+        /// </summary>
+        /// <param name="io_binding">Instance of OrtIoBinding</param>
+        /// <param name="name">UTF-8 zero terminated name</param>
+        /// <param name="mem_info">OrtMemoryInfo instance that contains device id. May be obtained from the device specific allocator instance</param>
+        /// <returns></returns>
+        public delegate IntPtr /* OrtStatus*/ DOrtBindOutputToDevice(IntPtr /*(OrtIoBinding)*/ io_binding, IntPtr /*(const char*) */ name, IntPtr /* const OrtMemoryInfo */ mem_info);
+        public static DOrtBindOutputToDevice OrtBindOutputToDevice;
+
+        /// <summary>
+        /// The function will return all bound output names in the order they were bound.
+        /// It is the same order that the output values will be returned after RunWithBinding() is used.
+        /// The function will allocate two native allocations  using the allocator supplied.
+        /// The caller is responsible for deallocating both of the buffers using the same allocator.
+        /// You may use OrtMemoryAllocation disposable class to wrap those allocations.
+        /// </summary>
+        /// <param name="io_binding">instance of OrtIoBinding</param>
+        /// <param name="allocator">allocator to use for memory allocation</param>
+        /// <param name="buffer">a continuous buffer that contains all output names.
+        /// Names are not zero terminated use lengths to extract strings. This needs to be deallocated.</param>
+        /// <param name="lengths">A buffer that contains lengths (size_t) for each of the returned strings in order.
+        /// You may use Marshal.ReadIntPtr() to read each element of the array. The buffer must be deallocated.</param>
+        /// <param name="count">this contains the count of names returned which is the number of elements in lengths.</param>
+        /// <returns></returns>
+        public delegate IntPtr /* OrtStatus*/ DOrtGetBoundOutputNames(IntPtr /* (const OrtIoBinding*) */ io_binding, IntPtr /* OrtAllocator* */ allocator,
+                                                                      out IntPtr /* char** */ buffer, out IntPtr /* size_t** */ lengths, out UIntPtr count);
+        public static DOrtGetBoundOutputNames OrtGetBoundOutputNames;
```

Comment:
>OrtGetBoundOutputNames [](start = 46, length = 22)

You seem to be exporting GetBoundOutputNames. Your earlier comment mentioned that you didn't.

Reply:
The earlier comment was about Input.